app/src/main/java/io/neurolab/main/TestModeActivity.java

```diff
@@ -48,6 +50,10 @@
     private int baudRate = 9600;
     private static final int ARDUINO_DEVICE_ID = 0x2341;
 
+    private static final int NUM_CHANNELS = 2;
```

Comment:
These should come from the values user set in Configurations. They are saved in sharedpreferences.

Reply:
@CloudyPadmal yes you are correct. Was going to do make it consistent throughout the app after your review.

Reply:
@CloudyPadmal If we want to get the shared preference instance of Settings Activity from this Activity(or any other activity) we need to use `getDefaultSharedPreferences(android.content.Context)` method because PrefFragmentCompat uses the same as Google Docs specify (_a context in the same package as this fragment._) It is an Activity context rather than an Application context which means its usage elsewhere would lead to references to that Activity even when its life cycle is not complete (because every reference usage to an Activity should have a tied lifecycle) which would in turn lead to memory leaks and problem with garbage collection. I am and was figuring out solutions for the same. Have a couple of solutions in my mind though not sure. Let me know how to proceed.

Reply:
@CloudyPadma Anyways give me some time. I will update it with the best possible solution available. Meanwhile, if you have any suggestions let me know.